### PR TITLE
Spotlight event headers

### PR DESCRIPTION
**Pull request by oetherington**
We've recently been added a text header by the site title to advertise the current event. This is configured using a database setting, but we often want to change the color of the text which is annoying as it requires code changes. We also generally show a spotlight at the same time.

Screenshot 2023-09-26 at 18 15 09

This PR makes two changes:

- It's now possible to configure the color of the text from the database setting
- Instead of using a database setting, it's now possible to configure the text directly as part of a spotlight (including setting the color):
Screenshot 2023-09-26 at 18 16 01

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205584785815210) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/spotlights/SpotlightEditorStyles.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ const styles = (theme: ThemeType): JssStyles => ({
       '& .input-spotlightImageId, & .input-darkImageId': {
         width: "100%"
       },
-      '& .input-documentId, & .input-documentType, & .input-position, & .input-draft, & .input-duration, & .input-customTitle, & .input-customSubtitle, & .input-draft, & .input-imageFade': {
+      '& .input-documentId, & .input-documentType, & .input-position, & .input-draft, & .input-duration, & .input-customTitle, & .input-customSubtitle, & .input-draft, & .input-imageFade, & .input-headerTitle, & .input-headerTitleLeftColor, & .input-headerTitleRightColor': {
```
Comment (oetherington) [inline packages/lesswrong/components/spotlights/SpotlightEditorStyles.tsx:19]: This is gross. C'est la vie.

File: packages/lesswrong/lib/collections/spotlights/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const getSpotlightUrl = ({document, documentType}: SpotlightDisplay) => {
```
Comment (oetherington) [inline packages/lesswrong/lib/collections/spotlights/helpers.ts:1]: This is refactored out of `SpotlightItem`

File: packages/lesswrong/components/common/HeaderEventSubtitle.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  title: string,
+  link: string,
+  background: string,
+}
+
+const useCurrentEvent = (): CurrentEvent | null => {
+  const currentEventSetting = currentEventHeader.get();
+  const spotlight = useCurrentFrontpageSpotlight();
+
+  if (currentEventSetting) {
```
Comment (Will-Howard) [inline packages/lesswrong/components/common/HeaderEventSubtitle.tsx:37]: I would honestly just remove the ability to do it this way. I expect we'll always have a spotlight to go along with events

File: packages/lesswrong/lib/reactRouterWrapper.tsx [hunk 4]
```diff
@@ -36,6 +36,7 @@ type LinkProps = {
   rel?: string
   eventProps?: Record<string, string>
   className?: string
+  style?: CSSProperties,
```
Comment (Will-Howard) [inline packages/lesswrong/lib/reactRouterWrapper.tsx:39]: Nice (doing the type properly)

**Review by Will-Howard — APPROVED** [review]
🦭

**Comment by jpaddison3** [comment]
Sweet

**Pull request merged**